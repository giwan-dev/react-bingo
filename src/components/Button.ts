import COLOR from 'colors';
import styled from 'styled-components';

const Button = styled.button`
  margin: 0;
  height: 35px;
  border: 0;
  border-radius: 5px;
  padding: 8px 16px;
  color: white;
  background-color: ${COLOR.primary};
  cursor: pointer;
  transition: opacity ease-out 0.2s;

  &:hover {
    opacity: 0.7;
  }
`;

export default Button;
