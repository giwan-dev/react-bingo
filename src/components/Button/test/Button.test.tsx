import React from 'react';
import { shallow } from 'enzyme';
import Button from '../Button';

describe('Button component', () => {
  it('should render without crashing', () => {
    const button = shallow((
      <Button />
    ));

    expect(button).toMatchSnapshot();
  });
});
