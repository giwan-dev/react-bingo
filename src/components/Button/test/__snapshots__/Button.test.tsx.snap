// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render without crashing 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  margin: 0;
  height: 35px;
  border: 0;
  border-radius: 5px;
  padding: 8px 16px;
  color: white;
  background-color: ",
          "#fd0d5c",
          ";
  cursor: pointer;
  transition: opacity ease-out 0.2s;

  &:hover {
    opacity: 0.7;
  }
",
        ],
      },
      "displayName": "styled.button",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "button",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
